Src/libCZI/SingleChannelAccessorBase.cpp

```diff
@@ -80,3 +81,61 @@ void CSingleChannelAccessorBase::CheckPlaneCoordinates(const libCZI::IDimCoordin
         }
     }
 }
+
+std::vector<int> CSingleChannelAccessorBase::CheckForVisibility(const libCZI::IntRect& roi, int count, const std::function<int(int)>& get_subblock_index) const
+{
+    std::vector<int> result;
+    result.reserve(count);
+
+    // handle the trivial cases
+    if (count == 0)
+    {
+        return result;
+    }
+    else if (count == 1)
+    {
+        result.push_back(0);
```

Comment:
You are I right I guess, that's actually a loophole, in this case the result would be wrong. This will not happen currently, all the rectangles we get here do intersect with the ROI, but this does not count as an argument for not fixing it of course. Will fix this.

Reply:
The cornerstone case you brought up is fixed in the refactored code, and a unittest for this case added.